backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanConfig.kt

```diff
@@ -115,17 +115,51 @@ class KonanConfig(val project: Project, val configuration: CompilerConfiguration
 
     internal val runtimeNativeLibraries: List<String> = mutableListOf<String>().apply {
         add(if (debug) "debug.bc" else "release.bc")
-        add(if (memoryModel == MemoryModel.STRICT) "strict.bc" else "relaxed.bc")
-        if (shouldCoverLibraries || shouldCoverSources) add("profileRuntime.bc")
-        if (configuration.get(KonanConfigKeys.ALLOCATION_MODE) == "mimalloc") {
-            if (!target.supportsMimallocAllocator()) {
+        val useMimalloc = if (configuration.get(KonanConfigKeys.ALLOCATION_MODE) == "mimalloc") {
+            if (target.supportsMimallocAllocator()) {
+                true
+            } else {
                 configuration.report(CompilerMessageSeverity.STRONG_WARNING,
                         "Mimalloc allocator isn't supported on target ${target.name}. Used standard mode.")
-                add("std_alloc.bc")
-            } else {
-                add("opt_alloc.bc")
-                add("mimalloc.bc")
+                false
+            }
+        } else {
+            false
+        }
+        val effectiveMemoryModel = when (memoryModel) {
+            MemoryModel.STRICT -> MemoryModel.STRICT
+            MemoryModel.RELAXED -> MemoryModel.RELAXED
+            MemoryModel.EXPERIMENTAL -> {
+                if (!useMimalloc) {
+                    configuration.report(CompilerMessageSeverity.STRONG_WARNING,
+                            "Experimental memory model requires mimalloc allocator. Used strict memory model.")
+                    MemoryModel.STRICT
```

Comment:
This behaviour unnecessarily complicates enabling experimental. Why not just use mimalloc if experimental "memory model" is selected? Is this actually user visible?

Reply:
Good idea, reworked.